Build failure for appstream-glib 0.7.0 on Mageia Cauldron (undefined references)

When trying to build appstream-glib 0.7.0 for Mageia Cauldron, I get build failures due to undefined references in the plugins. It looks something like this:

```
[77/83] cc  -o libappstream-builder/plugins/libasb_plugin_appdata.so 'libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o' -Wl,--as-needed -shared -fPIC -Wl,-soname,libasb_plugin_appdata.so -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -Wl,--as-needed -Wl,--no-undefined -Wl,-z,relro -Wl,-O1 -Wl,--build-id -Wl,--enable-new-dtags libappstream-builder/libappstream-builder.so.8.0.10 -Wl,--start-group -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wl,--end-group '-Wl,-rpath,$ORIGIN/:$ORIGIN/..:$ORIGIN/../../libappstream-glib' -Wl,-rpath-link,/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/libappstream-builder/plugins:/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/libappstream-builder:/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/libappstream-glib  
FAILED: libappstream-builder/plugins/libasb_plugin_appdata.so 
cc  -o libappstream-builder/plugins/libasb_plugin_appdata.so 'libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o' -Wl,--as-needed -shared -fPIC -Wl,-soname,libasb_plugin_appdata.so -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -Wl,--as-needed -Wl,--no-undefined -Wl,-z,relro -Wl,-O1 -Wl,--build-id -Wl,--enable-new-dtags libappstream-builder/libappstream-builder.so.8.0.10 -Wl,--start-group -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -Wl,--end-group '-Wl,-rpath,$ORIGIN/:$ORIGIN/..:$ORIGIN/../../libappstream-glib' -Wl,-rpath-link,/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/libappstream-builder/plugins:/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/libappstream-builder:/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/libappstream-glib  
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:76: undefined reference to `as_app_parse_file'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:80: undefined reference to `as_app_get_kind'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:81: undefined reference to `as_app_get_id'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:90: undefined reference to `as_app_validate'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:99: undefined reference to `as_problem_get_kind'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:100: undefined reference to `as_problem_get_message'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:100: undefined reference to `as_problem_kind_to_string'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:108: undefined reference to `as_app_get_icons'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:113: undefined reference to `as_app_get_project_license'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:114: undefined reference to `as_utils_is_spdx_license'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:132: undefined reference to `as_app_get_metadata_license'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:141: undefined reference to `as_utils_is_spdx_license'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:151: undefined reference to `as_app_get_update_contact'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:116: undefined reference to `as_utils_license_to_spdx'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:117: undefined reference to `as_utils_is_spdx_license'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:122: undefined reference to `as_app_set_project_license'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:162: undefined reference to `as_app_get_kind'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:171: undefined reference to `as_app_get_kind'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:182: undefined reference to `as_app_get_kind'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:184: undefined reference to `as_icon_new'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:185: undefined reference to `as_icon_set_kind'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:186: undefined reference to `as_icon_set_name'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:187: undefined reference to `as_app_add_icon'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:188: undefined reference to `as_app_add_category'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:189: undefined reference to `as_app_add_category'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:127: undefined reference to `as_app_set_project_license'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:173: undefined reference to `as_icon_new'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:174: undefined reference to `as_icon_set_kind'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:175: undefined reference to `as_icon_set_name'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:176: undefined reference to `as_app_add_icon'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:177: undefined reference to `as_app_add_category'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:178: undefined reference to `as_app_add_category'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:164: undefined reference to `as_icon_new'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:165: undefined reference to `as_icon_set_kind'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:166: undefined reference to `as_icon_set_name'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_process_filename':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:167: undefined reference to `as_app_add_icon'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_merge':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:253: undefined reference to `as_app_get_kind'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:256: undefined reference to `as_app_get_id'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `AS_APP':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-glib/as-app.h:48: undefined reference to `as_app_get_type'
libappstream-builder/plugins/asb_plugin_appdata@sha/asb-plugin-appdata.c.o: In function `asb_plugin_merge':
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:266: undefined reference to `as_app_get_kind'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:268: undefined reference to `as_app_get_id'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:272: undefined reference to `as_app_get_pkgname_default'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:271: undefined reference to `as_app_get_pkgname_default'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:274: undefined reference to `as_app_get_pkgname_default'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:274: undefined reference to `as_app_get_id'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:274: undefined reference to `as_app_add_veto'
/home/iurt/rpmbuild/BUILD/appstream-glib-0.7.0/build/../libappstream-builder/plugins/asb-plugin-appdata.c:279: undefined reference to `as_app_subsume_full'
collect2: error: ld returned 1 exit status
```

Interestingly, it builds fine on my local computer. It's just broken on the build system.

Full build log: 
[appstream-glib-0.7.0-build.0.20170723112421.log.txt](https://github.com/hughsie/appstream-glib/files/1168056/appstream-glib-0.7.0-build.0.20170723112421.log.txt)

--- a/libappstream-builder/plugins/meson.build
+++ b/libappstream-builder/plugins/meson.build
@@ -11,7 +11,7 @@ shared_module(
   dependencies : [
     gdkpixbuf,
   ],
-  link_with : asbuilder,
+  link_with : [asbuilder, asglib],
   c_args : asb_plugins_cargs,
   install : true,
   install_dir : plugindir,
@@ -28,7 +28,7 @@ shared_module(
   dependencies : [
     gdkpixbuf,
   ],
-  link_with : asbuilder,
+  link_with : [asbuilder, asglib],
   c_args : asb_plugins_cargs,
   install : true,
   install_dir : plugindir,
@@ -45,7 +45,7 @@ shared_module(
   dependencies : [
     gdkpixbuf,
   ],
-  link_with : asbuilder,
+  link_with : [asbuilder, asglib],
   c_args : asb_plugins_cargs,
   install : true,
   install_dir : plugindir,
@@ -62,7 +62,7 @@ shared_module(
   dependencies : [
     gdkpixbuf,
   ],
-  link_with : asbuilder,
+  link_with : [asbuilder, asglib],
   c_args : asb_plugins_cargs,
   install : true,
   install_dir : plugindir,
@@ -80,7 +80,7 @@ shared_module(
     gdkpixbuf,
     json_glib,
   ],
-  link_with : asbuilder,
+  link_with : [asbuilder, asglib],
   c_args : asb_plugins_cargs,
   install : true,
   install_dir : plugindir,
@@ -101,7 +101,7 @@ if get_option('enable-fonts')
       freetype,
       fontconfig,
     ],
-    link_with : asbuilder,
+    link_with : [asbuilder, asglib],
     c_args : asb_plugins_cargs,
     install : true,
     install_dir : plugindir,

